packages/@aws-cdk/aws-pipes-targets-alpha/lib/api-gateway.ts

```diff
@@ -0,0 +1,109 @@
+import { IInputTransformation, IPipe, ITarget, TargetConfig } from '@aws-cdk/aws-pipes-alpha';
+import { IRestApi } from 'aws-cdk-lib/aws-apigateway';
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+
+/**
+ * API Gateway REST API target properties.
+ */
+export interface ApiGatewayTargetParameters {
+  /**
+   * The input transformation to apply to the message before sending it to the target.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipetargetparameters.html#cfn-pipes-pipe-pipetargetparameters-inputtemplate
+   * @default - none
+   */
+  readonly inputTransformation?: IInputTransformation;
+
+  /**
+   * The method for API Gateway resource.
+   *
+   * @default '*' - ANY
+   */
+  readonly method?: string;
+
+  /**
+   * The path for the API Gateway resource.
+   *
+   * @default '/'
+   */
+  readonly path?: string;
+
+  /**
+   * The deployment stage for the API Gateway resource.
+   *
+   * @default - the value of `deploymentStage.stageName` of target API Gateway resource.
+   */
+  readonly stage?: string;
+
+  /**
+   * The headers to send as part of the request invoking the API Gateway REST API.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipetargethttpparameters.html#cfn-pipes-pipe-pipetargethttpparameters-headerparameters
+   * @default - none
+   */
+  readonly headerParameters?: Record<string, string>;
+
+  /**
+   * The path parameter values used to populate the API Gateway REST API path wildcards ("*").
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipetargethttpparameters.html#cfn-pipes-pipe-pipetargethttpparameters-pathparametervalues
+   * @default - none
+   */
+  readonly pathParameterValues?: string[];
+
+  /**
+   * The query string keys/values that need to be sent as part of request invoking the API Gateway REST API.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipetargethttpparameters.html#cfn-pipes-pipe-pipetargethttpparameters-querystringparameters
+   * @default - none
+   */
+  readonly queryStringParameters?: Record<string, string>;
+}
+
+/**
+ * A EventBridge Pipes target that sends messages to an EventBridge API destination.
+ */
+export class ApiGatewayTarget implements ITarget {
+  private restApi: IRestApi;
+  private restApiParameters?: ApiGatewayTargetParameters;
+  private restApiArn: string;
+  public readonly targetArn;
+
+  constructor(restApi: IRestApi, parameters?: ApiGatewayTargetParameters) {
+    this.restApi = restApi;
+    this.restApiParameters = parameters;
+
+    this.restApiArn = this.restApi.arnForExecuteApi(
+      this.restApiParameters?.method,
+      this.restApiParameters?.path || '/',
+      this.restApiParameters?.stage || this.restApi.deploymentStage.stageName,
```

Comment:
I think you might have an issue with `this.restApi.deploymentStage?.stageName` being undefined:

https://github.com/aws/aws-cdk/blob/ddbf57bd3bf4f8937a63c94d7188792ed37814b3/packages/aws-cdk-lib/aws-apigateway/lib/restapi.ts#L355-L361

Reply:
wait doesn't `!` mean it can't be `null` or `undefined` ?

Reply:
The non-null assertion operator is just a compile time override. Say you have this class:

```ts
class Example {
  value: string;
}
```

By default, TS will (rightfully) complain that `value` cannot be not a string: `Property 'value' has no initializer and is not definitely assigned in the constructor.(2564)`

You can override this check with the `!` operator, and `new Example().value` will be typed as a string, but will remain `undefined` at execution time

Reply:
I think the CDK automatically creates a stage though? So we should be good?

Reply:
I can see a couple of issues off the top of my head:

Given you're allowing `IRestApi` to construct `ApiGatewayTarget`, imported constructs can be passed. The `deploymentStage` will not be imported, even if the API has one:

```ts
const api = RestApi.fromRestApiId(this, 'RestApi', 'xxx');
assert(api.deploymentStage, 'Deployment stage is not defined');
```

You can also create a `RestApi` without a deployment with the `deploy` prop:

```ts
const api = new RestApi(this, 'api', {
  // Toggling this line will cause the assertion to fail  
  // deploy: false,
});
api.root.addMethod('GET');

assert(api.deploymentStage, 'Deployment stage is not defined');
```